From d15e9e8f9f332542815e085a990a28fe2219f21f Mon Sep 17 00:00:00 2001
From: neofang7 <neo.fang@intel.com>
Date: Wed, 31 May 2023 02:39:18 +0000
Subject: [PATCH] Condition check change of uvc camera id in jni.

Tencent Meeting cannot use uvc camera in Celadon ivi platform,
which caused by cameraId check in both jni and cameraservice.
CameraId is video device number + CameraIdOffset, CameraIdOffset
of uvc is 200, so it is definitely larger than the real camera
numbers and causes Invalid cameraId errors.

The fix contains two parts:
1. libcameraservice.so: change the condition check of cameraId,
iterate check app required cameraId with deviceIds;
2. Camera Jni: remove cameraId > NumberOfCameras condition.
Test: verify camera functionalities with meeting/camera apps.

Bug: OAM-109947
Signed-off-by: neofang7 <neo.fang@intel.com>
---
 core/jni/android_hardware_Camera.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/jni/android_hardware_Camera.cpp b/core/jni/android_hardware_Camera.cpp
index 365a18d174c9..cdb735ed811c 100644
--- a/core/jni/android_hardware_Camera.cpp
+++ b/core/jni/android_hardware_Camera.cpp
@@ -533,7 +533,7 @@ static void android_hardware_Camera_getCameraInfo(JNIEnv *env, jobject thiz,
     jint cameraId, jobject info_obj)
 {
     CameraInfo cameraInfo;
-    if (cameraId >= Camera::getNumberOfCameras() || cameraId < 0) {
+    if (cameraId < 0) {
         ALOGE("%s: Unknown camera ID %d", __FUNCTION__, cameraId);
         jniThrowRuntimeException(env, "Unknown camera ID");
         return;
-- 
2.17.1

